### PR TITLE
chore: update documentation on config parameters

### PR DESCRIPTION
**Pull request by ahuber21**
## Description

Add documentation about `use_raw_input` parameter. Also adds a missing explanation of `allow_sklearn_after_onedal` in one function, and the doc of all config parameters in the dictionary of default values in `onedal/_config.py`.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/_config.py [hunk 1]
```diff
@@ -63,6 +63,11 @@ def set_config(
         If True, allows to fallback computation to sklearn after onedal
         backend in case of runtime error on onedal backend computations.
         Global default: True.
+    use_raw_input : bool, default=None
+        If True, uses the raw input data in the onedal backend computations
+        without any checks on data consistency or validity.
+        Note: This option is not recommended for general use.
+        Global default: False.
     See Also
```
Comment (david-cortes-intel) [inline sklearnex/_config.py:71]: I think you need an empty line before starting a new section.

**Comment by david-cortes-intel** [comment]
@ahuber21 Since now the options differ from those in regular sklearn, would be ideal to add these to the .rst docs through autofunction - otherwise, it's very hard to discover these options or to notice that they differ from sklearn's.

**Comment by ahuber21** [comment]
> @ahuber21 Since now the options differ from those in regular sklearn, would be ideal to add these to the .rst docs through autofunction - otherwise, it's very hard to discover these options or to notice that they differ from sklearn's.

Hi @david-cortes-intel, yes these are valid concerns. We have decided to go that way because this functionality isn't intended for general use. In fact, we're hoping to remove it as quickly as possible. We don't want any adoption of this option, but rather implement array API, which makes all the raw input workarounds obsolete.

**Comment by ahuber21** [comment]
@david-cortes-intel thank you! Let me know if you agree with the latest version.

**Comment by david-cortes-intel** [comment]
Thanks for adding these. One detail though: 'Notes' is not a recognized section type, needs to be 'Note', otherwise the docs do not render correctly - see screenshot:
image

File: sklearnex/_config.py [hunk 2]
```diff
@@ -48,25 +50,46 @@ def set_config(
...
         If True, allows to fallback computation to host device
         in case particular estimator does not support the selected one.
         Global default: False.
+
     allow_sklearn_after_onedal : bool, default=None
         If True, allows to fallback computation to sklearn after onedal
         backend in case of runtime error on onedal backend computations.
         Global default: True.
+
+    use_raw_input : bool, default=None
```
Comment (david-cortes-intel) [inline sklearnex/_config.py:72]: Not entirely about this PR, but all of these should say 'bool or None', since 'None' is also supported.

File: sklearnex/_config.py [hunk 3]
```diff
@@ -90,25 +113,46 @@ def set_config(
...
     their previous values when the context manager is exited.
+
     See Also
     --------
     set_config : Set global scikit-learn configuration.
     get_config : Retrieve current values of the global configuration.
+
+    Warnings
+    --------
+    Using `use_raw_input=True` is not recommended for general use as it
```
Comment (david-cortes-intel) [inline sklearnex/_config.py:154]:
```suggestion
    Using ``use_raw_input=True`` is not recommended for general use as it
```

File: sklearnex/_config.py [hunk 4]
```diff
@@ -48,25 +50,46 @@ def set_config(
...
+    their previous values when the context manager is exited.
+
     See Also
     --------
     config_context : Context manager for global configuration.
     get_config : Retrieve current values of the global configuration.
+
+    Warnings
+    --------
+    Using `use_raw_input=True` is not recommended for general use as it
```
Comment (david-cortes-intel) [inline sklearnex/_config.py:91]:
```suggestion
    Using ``use_raw_input=True`` is not recommended for general use as it
```

File: sklearnex/_config.py (hunk 2 above)
Comment (ahuber21) [reply sklearnex/_config.py:72]: Yes I agree. We are not at all consistent here.

**Comment by ahuber21** [comment]
> Thanks for adding these. One detail though: 'Notes' is not a recognized section type, needs to be 'Note', otherwise the docs do not render correctly - see screenshot: ...

Hope it's better now!

**Review by david-cortes-intel — APPROVED** [review]
